### DISCUSSION TITLE
SSO: getSession in azure ad how we handle server side and client side auth?

### DISCUSSION THREAD

**Discussion by manoj-mukherjee-maersk** [nextauthjs/next-auth#3218] [Help]
### Question 💬

On page load we need to automatically redirect to azure-ad login page. How will do SSO azure-ad login in serverSideProps?

session always coming null. 

<img width="611" alt="Screenshot 2021-11-16 at 11 27 53 PM" src="https://user-images.githubusercontent.com/88817537/142039880-44b49dbe-7cd1-4012-9796-95a70db83d0f.png">

### How to reproduce ☕️

```js 
// pages/index.js
export async function getServerSideProps(ctx) {
  const session = await getSession(ctx);
  console.log({ session });
  if (!session) {
    return {
      redirect: {
        destination: "/api/auth/signin",
        permanent: false,
      },
    };
  }
  return {
    props: {
      session: session,
    },
  };
}
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
Looks like a misconfiguration at first glance. That endpoint isn't supposed to return HTML (which seems to be the case judging from the error).

Have you set `NEXTAUTH_URL`, if yes to what? Can you show your file structure and the content of `pages/api/[...nextauth].js` file?

**Reply by manoj-mukherjee-maersk** [reply]
Yes we set NEXTAUTH_URL in .env.local file 
```js
NEXTAUTH_URL=http://localhost:3000
```

. clientSide useSession working fine. In server side rendering we are not getting session.

<img width="300" alt="Screenshot 2021-11-17 at 10 39 31 AM" src="https://user-images.githubusercontent.com/88817537/142138763-4b2281f6-5ff5-44a8-bffc-a04bb2bbb2d2.png">
content [...nextauth].ts file
<img width="658" alt="Screenshot 2021-11-17 at 10 40 17 AM" src="https://user-images.githubusercontent.com/88817537/142138809-e10413ff-2261-4b52-84b2-e22ef9913158.png">

**Reply by manoj-mukherjee-maersk** [reply]
Could you please add azure-ad example with server side session.

**Reply by balazsorban44** [reply]
I don't think there is anything special about Azure AD here, to be honest.

**Reply by manoj-mukherjee-maersk** [reply]
Any suggestions to resolved getSession null issue?

**Reply by balazsorban44** [reply]
Sorry, I haven't found anything. I could not reproduce this locally. Check that you don't block any cookies, try different browsers/devices and see if they behave differently.

**Reply by manoj-mukherjee-maersk** [reply]
I tried but no luck.

**Reply by manoj-mukherjee-maersk** [reply]
Its resolved with latest version

**Reply by fredericoo** [reply]
Also facing issue.
It's very intermittent to me. On my computer it almost never works, sometimes resetting the database and server fixes it, sometimes not. 

On my teammates computer it always works. 

Perhaps it's got something to do with node versions?

**Reply by manoj-mukherjee-maersk** [reply]
@fredericoo  upgrade node 16.13+ LTS version with latest nextjs its works. In older nodejs version its issue.

**Reply by manoj-mukherjee-maersk** [reply]
In production deployed in linux system. In server side session again null. Its deployed with docker image. any solution for docker image. NEXTAUTH_URL setting as host. @balazsorban44

**Reply by manoj-mukherjee-maersk** [reply]
@balazsorban44 
In Local also session null for api route. But getting session  getServerSideProps. same getSession({req}) giving null inside api route.

    "next": "^12.1.0",
    "next-auth": "^4.2.1",
    "react": "^17.0.2",

node -> v16.13.2
OS - Mac OS M1 chip Big sure 11.3.1

session got
<img width="489" alt="Screenshot 2022-02-24 at 1 33 21 AM" src="https://user-images.githubusercontent.com/88817537/155399106-4f8bddfc-a455-4b73-ac42-a9cf01cb6481.png">

session -> null inside api route
<img width="764" alt="Screenshot 2022-02-24 at 1 34 48 AM" src="https://user-images.githubusercontent.com/88817537/155399240-ff792648-21f7-4b2c-b512-ef8dd7d90810.png">

<img width="585" alt="Screenshot 2022-02-24 at 1 29 38 AM" src="https://user-images.githubusercontent.com/88817537/155398333-2ae9470b-6a84-4261-8dad-c1155c73a2b7.png">
